packages/sdk/src/clients/optimisticOracle/client.ts

```diff
@@ -24,36 +24,49 @@ export enum RequestState {
   Settled, // Final price has been set in the contract (can get here from Expired or Resolved).
 }
 
+// all events have these values
 export type RequestKey = {
   requester: string;
   identifier: string;
   timestamp: number;
   ancillaryData: string;
 };
-export type Request = RequestKey & {
-  proposer: string;
-  disputer: string;
-  currency: string;
-  settled: boolean;
-  refundOnDispute: boolean;
-  proposedPrice: string;
-  resolvedPrice: string;
-  expirationTime: string;
-  reward: string;
-  finalFee: string;
-  bond: string;
-  customLiveness: string;
-  price: string;
-  payout: string;
-  state: RequestState;
-};
+export type Request = RequestKey &
+  // this is partial since we dont know what events we have to populate parts of this
+  Partial<{
```

Comment:
just my opinion, not requesting a change, but is there any field that will always be there on all requests? I'm asking because `Partial` is particularly annoying for consumers as you need to check every field for possible undefined

Reply:
the only fields guaranteed to be there are the "request key" im calling it, which are the properties required to uniquely identify the request: identifier, requester, timestamp, ancillary data. The rest have to be partial because they are dependent on the set of events we have, and unfortunately due to querying constraints on providers, we cant have the whole set of events quickly.  I agree this is really annoying to deal with, but i just dont see a way around this at the moment, so better to make this explicit in the type.